src/test/javascript/spec/component/overview/exercise-details/course-exercise-details.component.spec.ts

```diff
@@ -211,24 +210,24 @@ describe('CourseExerciseDetailsComponent', () => {
         studentParticipation.exercise = exercise;
 
         const exerciseDetail = { ...exercise, studentParticipations: [studentParticipation] };
-        const exerciseDetailReponse = of({ body: exerciseDetail });
+        const exerciseDetailResponse = of({ body: exerciseDetail });
 
         // return initial participation for websocketService
-        stub(participationWebsocketService, 'getParticipationForExercise').returns(studentParticipation);
-        stub(complaintService, 'findBySubmissionId').returns(of({} as EntityResponseType));
+        jest.spyOn(participationWebsocketService, 'getParticipationForExercise').mockReturnValue(studentParticipation);
+        jest.spyOn(complaintService, 'findBySubmissionId').mockReturnValue(of({} as EntityResponseType));
 
-        mergeStudentParticipationStub.returns(studentParticipation);
+        mergeStudentParticipationMock.mockReturnValue(studentParticipation);
         const changedParticipation = cloneDeep(studentParticipation);
         const changedResult = { ...result, id: 2 };
         changedParticipation.results = [changedResult];
-        subscribeForParticipationChangesStub.returns(new BehaviorSubject<Participation | undefined>(changedParticipation));
+        subscribeForParticipationChangesMock.mockReturnValue(new BehaviorSubject<Participation | undefined>(changedParticipation));
```

Comment:
Does this override the mock definition from above or does it append one mock value?

Reply:
It appends a value, from what I can tell